public/video-ui/src/actions/VideoActions/saveVideo.ts

```diff
@@ -0,0 +1,56 @@
+import VideosApi, { Video } from '../../services/VideosApi';
+import { showError } from '../../slices/error';
+import { setSaving } from '../../slices/saveState';
+import { fetchUsages, updateVideoUsageWebTitle, UsageData } from '../../slices/usage';
+import { setVideo } from '../../slices/video';
+import { AppDispatch } from '../../util/setupStore';
+
+const errorMessages = {
+  saveVideoDefault: 'Could not save video',
+  saveVideoConflict: 'Could not save video as another user (or tab) is currently editing this video',
+  fetchUsagesRejected: 'Could not fetch usages to update canonical page',
+  canonicalPageUpdate: 'Could not update canonical page'
+};
+
+export const saveVideo = (video: Video) => async (dispatch: AppDispatch) => {
+
+  dispatch(setSaving(true));
+  dispatch(setVideo(video));
+
+  const savedVideo: Video | undefined = await VideosApi.saveVideo(video.id, video)
+    .catch(error => {
+      const message = error.status === 409
+        ? errorMessages.saveVideoConflict
+        : errorMessages.saveVideoDefault;
+      dispatch(showError(message, error));
+      return undefined;
+    });
+  if (!savedVideo) {
+    dispatch(setSaving(false));
+    return;
+  }
+
+  const fetchUsagesAction = await dispatch(fetchUsages(video.id));
+  if (fetchUsagesAction.meta.requestStatus !== 'fulfilled') {
```

Comment:
Looks like the idiomatic RTK way to handle this is using the unwrap function: 
https://redux-toolkit.js.org/api/createAsyncThunk#unwrapping-result-actions. It would make it a little more consistent with how we are handling promises in the rest of this function.